includes/multi-currency/MultiCurrency.php

```diff
@@ -1441,28 +1441,55 @@ public function is_multi_currency_settings_page(): bool {
 		);
 	}
 
+	/**
+	 * Function used to compute the customer used currencies, used as internal callable for get_all_customer_currencies function.
+	 *
+	 * @return array
+	 */
+	public function callable_get_customer_currencies() {
+		global $wpdb;
+
+		$currencies  = $this->get_available_currencies();
+		$query_union = [];
+
+		if ( class_exists( 'Automattic\WooCommerce\Utilities\OrderUtil' ) &&
+					\Automattic\WooCommerce\Utilities\OrderUtil::custom_orders_table_usage_is_enabled() ) {
+			foreach ( $currencies as $currency ) {
+				$query_union[] = sprintf(
+					'SELECT "%1$s" AS currency_code, EXISTS(SELECT currency FROM %2$s WHERE currency="%1$s" LIMIT 1) AS exists_in_orders',
```

Comment:
@ovidiul, out of curiosity, did we check with `EXPLAIN` which indexes are used in this query? Can you share perhaps a sample output?

Reply:
If I read correctly the [WC migration](https://github.com/woocommerce/woocommerce/blob/c8b2305dcd5354c284e120f00ade43290c3b7723/plugins/woocommerce/src/Database/Migrations/CustomOrderTable/PostToOrderTableMigrator.php#L26), it was no index added.

As no indexes are available, we can not do much here. If you get a bit of time, please file a WC issue to highlight that indexes might be missing for HPOS tables.